### PR TITLE
removes bugs from select_multiple section of readreplace

### PR DESCRIPTION
**Pull request by kelseylarson**
Adds option to readreplace to drop observations using the corrections file; removes miscellaneous bugs from the selectmultiple section.

### CODE REVIEW

File: ado/ipacheckreadreplace.ado [hunk 1]
```diff
@@ -83,16 +93,31 @@ qui {
...
-		error 498
-	}
 	
-	tempfile selectmult allcorrections
+	***************** remove entries that should be dropped ********************
+	tempfile selectmult allcorrections droplist
 	save `allcorrections'
-
+	if !missing("`drop'") {
+		cap tostring `drop', replace
```
Comment (boyercb) [inline ado/ipacheckreadreplace.ado]: Why is this result being captured? Are you just trying to abbreviate the longer, but more precise code which would be:
```
cap confirm string variable `drop'
if _rc {
    tostring `drop', replace
}
```
I am concerned by just capturing the result of tostring directly you may be preventing a more informative error message.

File: ado/ipacheckreadreplace.ado [hunk 2]
```diff
@@ -83,16 +93,31 @@ qui {
...
+		replace `drop' = lower(strtrim(`drop'))
```
Comment (boyercb) [inline ado/ipacheckreadreplace.ado]: So here for the drop column... Are we saying that the user needs to write out "drop" in the drop column? A more natural fit to me would seem to be "yes" or "1" or something?

File: ado/ipacheckreadreplace.ado [hunk 3]
```diff
@@ -83,16 +93,31 @@ qui {
...
+		save `allcorrections', replace
+		keep if `drop' == "drop"
+		keep `drop' `id'
+		count
```
Comment (boyercb) [inline ado/ipacheckreadreplace.ado]: Simpler code (without displayed result):

```
if _N > 0 {
 ...
}
```

File: ado/ipacheckreadreplace.ado [hunk 4]
```diff
@@ -83,16 +93,31 @@ qui {
...
+			drop `drop_merge' `drop'
+			preserve
+		}
+		use `allcorrections', clear
+		drop if `drop' == "drop"
+		save `allcorrections', replace
+	}
+	******************* split selectmultiples and other corrections ************
 	keep if !missing(`selectmultiple')
 	count
```
Comment (boyercb) [inline ado/ipacheckreadreplace.ado]: See comment about simplifying this code above.

File: ado/ipacheckreadreplace.ado [hunk 5]
```diff
@@ -110,21 +135,20 @@ qui {
...
 			noi list `id' if `mult_merge' == 2
 		}
 		drop if `mult_merge' == 2
 		drop `mult_merge'
-		
-		* replace the selectmultiples
+
 		tempvar concat
-		foreach var in `correction_vars' {
+		foreach var in `"`correction_vars'"' {
```
Comment (boyercb) [inline ado/ipacheckreadreplace.ado]: Are you expecting a nested list here? Why did you need compound quotes?

File: ado/ipacheckreadreplace.ado [hunk 6]
```diff
@@ -148,10 +172,18 @@ qui {
...
+	use `allcorrections', clear
+	count 
+	if `r(N)' > 0 {
+		restore
+		readreplace using `allcorrections', ///
 			id(`id') variable(`variable') value(`value') use
-	
+	}
+	else {
+		restore
```
Comment (boyercb) [inline ado/ipacheckreadreplace.ado]: I'm going to have to think about this for a minute but it strikes me that in it's current form there seems to be too many `preserve` and `restore` statements. I wonder if a different logical arrangement might lead to less of these pesky calls to create temporary files?

File: ado/ipacheckreadreplace.ado (hunk 1 above)
Comment (kelseylarson) [inline ado/ipacheckreadreplace.ado]: Yes, that would be the better option!

File: ado/ipacheckreadreplace.ado [hunk 7]
```diff
@@ -44,31 +44,41 @@ qui {
...
-		local using subinstr("`using'", "using", "", 1)
-		use `using', `import' clear
+		use "`using'", `import' clear
 	}
+	if _N == 0 {
+		di "no replacements"
+		exit
+	}
+	
+	cap assert !missing(`id')
```
Comment (kelseylarson) [inline ado/ipacheckreadreplace.ado]: This is before we hit the readreplace command, and it makes sure that the form has what it needs (ie, nonmissing ids) for the "drop" section and the "select multiple" section as well.

File: ado/ipacheckreadreplace.ado (hunk 2 above)
Comment (kelseylarson) [inline ado/ipacheckreadreplace.ado]: My main concern is that someone might confuse columns and accidentally drop something that they didn't mean to drop. Should the relative ease of "yes" or "1" take precedent over the accidental drop protection of "drop"? I could see merits to either one.

File: ado/ipacheckreadreplace.ado (hunk 3 above)
Comment (kelseylarson) [inline ado/ipacheckreadreplace.ado]: Yeah, makes sense.

File: ado/ipacheckreadreplace.ado (hunk 5 above)
Comment (kelseylarson) [inline ado/ipacheckreadreplace.ado]: This was part of my attempt to fix one of the odder errors I was getting, but it might not be necessary.
